PWGHF/HFC/TableProducer/correlatorDMesonPairs.cxx

```diff
@@ -0,0 +1,636 @@
+///
+/// \author Fabio Colamaria <fabio.colamaria@ba.infn.it>, INFN Bari
+/// \author Andrea Tavira García <tavira-garcia@ijclab.in2p3.fr>, IJCLab Orsay
+
+#include "Common/Core/TrackSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "Framework/runDataProcessing.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod::hf_cand;
+using namespace o2::aod::hf_cand_2prong;
+using namespace o2::aod::hf_cand_3prong;
+using namespace o2::constants::math;
+
+///
+/// Returns deltaPhi value in range [-pi/2., 3.*pi/2], typically used for correlation studies
+///
+double getDeltaPhi(double phiD, double phiDbar)
+{
+  return RecoDecay::constrainAngle(phiDbar - phiD, -o2::constants::math::PIHalf);
+}
+
+namespace
+{
+enum CandidateTypeSel {
+  SelectedD = 0, // This particle is selected as a D
+  SelectedDbar,  // This particle is selected as a Dbar
+  TrueD,         // This particle is a true D
+  TrueDbar       // This particle is a true Dbar
+};
+} // namespace
+
+// histogram binning definition
+const int massAxisBins = 120;
+const double massAxisMin = 1.5848;
+const double massAxisMax = 2.1848;
+const int phiAxisBins = 32;
+const double phiAxisMin = 0.;
+const double phiAxisMax = o2::constants::math::TwoPI;
+const int yAxisBins = 100;
+const double yAxisMin = -5.;
+const double yAxisMax = 5.;
+const int ptDAxisBins = 180;
+const double ptDAxisMin = 0.;
+const double ptDAxisMax = 36.;
+
+using McParticlesPlus2Prong = soa::Join<aod::McParticles, aod::HfCand2ProngMcGen>;
+using McParticlesPlus3Prong = soa::Join<aod::McParticles, aod::HfCand3ProngMcGen>;
+
+struct HfCorrelatorDMesonPairs {
+  SliceCache cache;
+  Preslice<aod::HfCand2Prong> perCol2Prong = aod::hf_cand::collisionId;
+  Preslice<aod::HfCand3Prong> perCol3Prong = aod::hf_cand::collisionId;
+
+  Produces<aod::D0Pair> entryD0Pair;
+  Produces<aod::D0PairRecoInfo> entryD0PairRecoInfo;
+  Produces<aod::DPlusPair> entryDPlusPair;
+  Produces<aod::DPlusPairRecoInfo> entryDPlusPairRecoInfo;
+
+  Configurable<bool> verbose{"verbose", false, "Enable debug mode"};
+  // Selection Flags
+  Configurable<int> selectionFlagD0{"selectionFlagD0", 1, "Selection Flag for D0"};
+  Configurable<int> selectionFlagD0bar{"selectionFlagD0bar", 1, "Selection Flag for D0bar"};
+  Configurable<int> selectionFlagDplus{"selectionFlagDPlus", 1, "Selection Flag for DPlus"};
+  // Kinematics
+  Configurable<double> etaCutMax{"etaCutMax", 4.0, "max. eta accepted"};
+  Configurable<double> etaCutMin{"etaCutMin", -4.0, "min. eta accepted"};
+  Configurable<double> dcaXYCut{"dcaXYCut", 0.0025, "cut in dcaXY"};
+  Configurable<double> dcaZCut{"dcaZCut", 0.0025, "cut in dcaZ"};
```

Comment:
Do not use the word `Cut`. Instead use `Min`, `Max`, to clearly indicate the type of criterion, and mention it in the description.